Fix parsing enums with items evaluating to 0

mlocati opened the pull request:
I have a typescript like this:

```typescript
enum MyEnum {
    Name0 = 0,
    Name1 = 1,
    Name2 = 2,
}
interface MyInterface
{
    fieldName: MyEnum 
}
const decoder = JsonDecoder.objectStrict<MyInterface>(
    {
        fieldName: JsonDecoder.enumeration<MyEnum>(MyEnum , 'MyEnum'),
    },
    'MyDecoder'
);
```

When checking a json like this:

```json
{
   "fieldName": 0
}
```

The code fails, since the result of `Object.values(enumObj).find((x: any) => x === json);` is `0`.

Since `find` returns `undefined` if the value can't be found, what about using it?

joanllenas commented:
Hey @mlocati , good catch! 
Sounds reasonable to me 👍 
Would you mind adding a test for the bug fix?

mlocati commented:
> Would you mind adding a test for the bug fix?

Done.
I've added commit 89f9991338d12eef2a00775a448d1d5a67c2b4dd.
It makes tests fail:

```
$ npm run test

  json-decoder
    [...omissis...]
    enumeration
      1) should decode when the value is in the enum
      √ should fail when the value is not in the enum
    [...omissis...]

  89 passing (145ms)
  1 failing

  1) json-decoder
       enumeration
         should decode when the value is in the enum:
     AssertionError: expected { Object (error) } to be an instance of Ok
      at expectOkWithValue (src\json-decoder.spec.ts:10:37)
      at Context.<anonymous> (src\json-decoder.spec.ts:121:7)
      at processImmediate (node:internal/timers:466:21)
```

Then I added aade05e4755b25f11526ec137f9dbe95086cb03b which fixes the issue.

What about adding a GitHub Action that run the tests?

joanllenas commented:
Awesome, thanks @mlocati !
I will merge / create the patch release later today.

mlocati commented:
> What about adding a GitHub Action to run the tests?

Maybe something like #33?

joanllenas commented:
Thanks for catching this and contributing with the PR @mlocati !
Released here: https://github.com/joanllenas/ts.data.json/releases/tag/v2.1.1
